src/macros.js

```diff
@@ -214,6 +216,29 @@ defineMacro("\\show", (context) => {
     return '';
 });
 
+//////////////////////////////////////////////////////////////////////
+// conditionals
+defineMacro("\\else", function(context) {
+    // When \else is expanded, TeX reads to the end of any text that
+    // ought to be skipped.
+    if (!context.conditions[context.conditions.length - 1]) {
+        throw new ParseError("Extra \\else.");
+    }
+    context.skipConditionalText();
+    return '';
+});
+defineMacro("\\fi", function(context) {
+    if (context.conditions.length === 0) {
+        throw new ParseError("Extra \\fi.");
+    }
+    context.conditions.pop();
+    return '';
+});
+
+defineConditional("\\iftrue", () => true);
+defineConditional("\\iffalse", () => false);
+defineConditional("\\ifmmode", (context) => context.mode === 'math');
```

Comment:
I seem to recall when implementing `\TextOrMath` that this wasn't possible before. I'm guessing it works now though.

Here's an attempt at `\newif`:

```js
defineMacro("\\newif", function(context) {
    let value = false;
    const name = context.popToken();  // don't expand following token
    // LaTeX silently strips off the first two characters, intending them to be 'if'. This is a more strict check:
    if (!name.text.startsWith('\\if')) {
        throw new ParseError("\newif followed by something other than \if...");
    }
    const cond = name.text.slice(2);
    defineConditional(name.text, () => value);
    defineMacro(`\\${cond}false`, () => {
        value = false;
        return '';
    });
    defineMacro(`\\${cond}true`, () => {
        value = true;
        return '';
    });
    return '';
});
```